### PR TITLE
1063 encounter report

### PR DESCRIPTION
**Pull request by clemens-msupply**
Add required work to get data for the encounters report.

The report is temp configured as a Patient report, i.e. to test it you can print it from the patient print button.

Also fixes an edge case for program events.

Part of #1063

### CODE REVIEW

File: server/service/src/sync/program_schemas/encounters_arg_schema.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "EncounterStatus": {
+      "enum": ["SCHEDULED", "COMPLETED", "CANCELLED"],
```
Comment (mark-prins) [inline server/service/src/sync/program_schemas/encounters_arg_schema.json:5]: I've created a new database and these are the values for the encounter status:

Screenshot 2023-02-13 at 11 58 28 AM

and if I try to print I get this

Screenshot 2023-02-13 at 11 55 23 AM
 
because of this

```
"details": "DBError { msg: \"UNKNOWN\", extra: \"\\\"\\\\\\\"invalid input value for enum encounter_status: \\\\\\\\\\\\\\\"COMPLETED\\\\\\\\\\\\\\\"\\\\\\\"\\\"\" }"
```

Comment (mark-prins) [reply server/service/src/sync/program_schemas/encounters_arg_schema.json:5]: I've changed the `encounter_status` type in the database directly in order to test.
I'd suggest changing the type - that would make it consistent with the hiv_care and hiv_testing encounters too. Also, I like 'COMPLETED' better than 'DONE' 😉

File: server/service/src/sync/program_schemas/encounters_report.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (mark-prins) [inline server/service/src/sync/program_schemas/encounters_report.json:1]: :nit

the orange borders don't do it for me..

image

it's an example though, right? So no need to change. I'm just being whiney.

**Review by mark-prins — CHANGES_REQUESTED** [review]
😁  Looking good! There's just a subjective change and a postgres (probably) issue.

File: client/packages/programs/src/JsonForms/common/components/Date.tsx [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+              !!props.error ? { sx: { color: 'error.main' } } : undefined
+            }
+            {...textInputProps}
+          />
+        );
+      }}
+      {...props}
+    />
+  );
+};
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/common/components/Date.tsx]: thanks!

File: server/service/src/sync/program_schemas/encounters_arg_schema.json (hunk 1 above)
Comment (clemens-msupply) [reply server/service/src/sync/program_schemas/encounters_arg_schema.json:5]: Yes Completed sounds better, was Carl's idea to rename it :)

Will fix it!

File: server/service/src/sync/program_schemas/encounters_report.json (hunk 2 above)
Comment (clemens-msupply) [reply server/service/src/sync/program_schemas/encounters_report.json:1]: Yes its just about getting all the data right now! I still need to do the style it.

File: server/graphql/programs/src/types/encounter.rs [hunk 4]
```diff
@@ -178,16 +201,21 @@ impl EncounterNode {
...
+        let mut program_filter = filter
+            .as_ref()
             .map(|f| f.to_domain())
             .unwrap_or(ProgramEventFilter::new())
-            .patient_id(EqualFilter::equal_to(&self.encounter_row.patient_id))
-            .document_type(EqualFilter::equal_to(&self.encounter_row.r#type))
-            .document_name(EqualFilter::equal_to(&self.encounter_row.name));
+            .patient_id(EqualFilter::equal_to(&self.encounter_row.patient_id));
+        program_filter =
+            program_filter.document_type(EqualFilter::equal_to(&self.encounter_row.r#type));
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply server/graphql/programs/src/types/encounter.rs:214]: Yes don't know why I did that :)

File: server/service/src/sync/program_schemas/encounters_arg_schema.json (hunk 1 above)
Comment (clemens-msupply) [reply server/service/src/sync/program_schemas/encounters_arg_schema.json:5]: Ok there where a couple of other issues as well, thought I ran the test but maybe not :(

**Review by mark-prins — APPROVED** [review]
🎉  Looking good - thanks!
Just had to fix a linting issue, I've committed in b4df0057e7f8ae04dc0934ea0a8fcf533c503662
